sparkle/src/components/Breadcrumbs.tsx

```diff
@@ -1,56 +1,113 @@
 import type { ComponentType } from "react";
 import React from "react";
 
-import { SparkleContextLinkType } from "@sparkle/context";
-import { ChevronRightIcon, Icon, SparkleContext } from "@sparkle/index";
+import { noHrefLink, SparkleContextLinkType } from "@sparkle/context";
+import {
+  ChevronRightIcon,
+  DropdownMenu,
+  Icon,
+  SparkleContext,
+  Tooltip,
+} from "@sparkle/index";
+
+const LABEL_TRUNCATE_LENGTH_MIDDLE = 15;
+const LABEL_TRUNCATE_LENGTH_END = 30;
+
+type BreadcrumbItem = {
+  label: string;
+  icon?: ComponentType<{ className?: string }>;
+  href?: string;
+};
 
 type BreadcrumbProps = {
-  items: {
-    label: string;
-    icon?: ComponentType<{ className?: string }>;
-    href?: string;
-  }[];
+  items: BreadcrumbItem[];
 };
 
 export function Breadcrumbs({ items }: BreadcrumbProps) {
   const { components } = React.useContext(SparkleContext);
 
-  const Link: SparkleContextLinkType = components.link;
+  const { itemsShown, itemsHidden } = items.reduce(
+    (acc, item, index) => {
+      if (items.length <= 5 || index < 2 || index >= items.length - 2) {
+        acc.itemsShown.push(item);
+      } else if (index === 2) {
+        acc.itemsShown.push({ label: "..." });
+      } else {
+        acc.itemsHidden.push(item);
+      }
+      return acc;
+    },
+    { itemsShown: [] as typeof items, itemsHidden: [] as typeof items }
```

Comment:
To avoid the `as` you could type acc: 

```js
type BreadcrumbsAccumulator = {
  itemsShown: BreadcrumbItem[];
  itemsHidden: BreadcrumbItem[];
};


  const { itemsShown, itemsHidden } = items.reduce(
    (acc: BreadcrumbsAccumulator, item, index) => {
      if (items.length <= 5 || index < 2 || index >= items.length - 2) {
        acc.itemsShown.push(item);
      } else if (index === 2) {
        acc.itemsShown.push({ label: "..." });
      } else {
        acc.itemsHidden.push(item);
      }
      return acc;
    },
    { itemsShown: [], itemsHidden: [] }
  );
```